### PR TITLE
Add extra components for molecule input and viewing

### PR DESCRIPTION
**Pull request by AdrianDAlessandro**
This PR creates a few dash components to allow for extra ways to input chemical structures, as well as viewing them. This includes"
 - Uploading a .mol file
 - Viewing a chemical structure image (from uploaded .mol file)
 - Drawing a chemical structure and getting the SMILES string

Currently, they are just added to the bottom of the existing dash app.

### CODE REVIEW

**Comment by AdrianDAlessandro** [comment]
There is a dependancy version issue with the `dash-bio` package required for the Jsme molecular drawer for these models. This will be a problem for any of the other models that are locked in to `scikit-learn<0.20`. We will need to be flexible with this.

**Comment by cc-a** [comment]
Hmm... this is the sort of thing that makes separation of the front-end (serving the dash app) and the back-end (running the model) compelling.

**Comment by AdrianDAlessandro** [comment]
That's a good point @cc-a 

Although I don't have any idea how you would make that possible

**Comment by AdrianDAlessandro** [comment]
> Hmm... this is the sort of thing that makes separation of the front-end (serving the dash app) and the back-end (running the model) compelling.

After some discussion, the decision is to stick with the current approach where front-end and back-end are in the one container. This will be simpler for the researchers to work with.

We will work around this import issue by including the dash-bio objects in a separate module that can be optionally used (i.e. `advanced_widgets`)

**Comment by AdrianDAlessandro** [comment]
This is ready to be reviewed. It doesn't contain the JSME drawer component, I will have a separate PR for that. The idea is that new apps can be developed by editing these files:
 - `layout.py` - The structure of the html/bootstrap components
 - `callbacks.py` - The interactivity of the components
 - `model.py` - The backend model

There are also some more complicated components in `components.py` that related to uploading a `.mol` file and displaying an image of the molecule. They are not very flexible at the moment, but they're usable and repeatable.

File: README.md [hunk 1]
```diff
@@ -4,17 +4,17 @@ A platform for sharing trained machine learning models for chemistry.
...
+A [Docker image of the containerised app](https://github.com/ImperialCollegeLondon/SupraShare/pkgs/container/suprashare%2Fsuprashare) is automatically generated [using GitHub actions](https://github.com/ImperialCollegeLondon/SupraShare/blob/develop/.github/workflows/ci.yml) when changes are merged into the main branch.
 
-The image is added to the webserver at [suprashare.rcs.ic.ac.uk](https://suprashare.rcs.ic.ac.uk) by manually updating (i) the [docker-compose.yml file](https://docs.docker.com/compose/compose-file/) (ii) the [reverse proxy config](https://caddyserver.com/docs/caddyfile) and (iii) the static index page (front page of the web site), all within the virtual machine running the webserver. 
+The image is added to the webserver at [suprashare.rcs.ic.ac.uk](https://suprashare.rcs.ic.ac.uk) by manually updating (i) the [docker-compose.yml file](https://docs.docker.com/compose/compose-file/) (ii) the [reverse proxy config](https://caddyserver.com/docs/caddyfile) and (iii) the static index page (front page of the web site), all within the virtual machine running the webserver.
```
Comment (dandavies99) [inline README.md:12]:
```suggestion

## Template structure

The code in this template is separated into several submodules depending on its function. 

TODO: 1 sentence/bullet point per file describing its purpose. 

```mermaid
  graph TD;
      Components-->Layout;
      Model-->Callbacks;
      Callbacks-->App;
      Layout-->App;
```
```

File: website/layout.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+
+def content() -> list:
+    """Generate the content for the webpage.
+
+    Returns:
+        list: A list of each component ordered from the top of the page to the bottom.
+    """
+    # Read the static markdown content in static/index.md for the top of the webpage.
+    # Edit this file to edit the text content at the top of the page.
```
Comment (dandavies99) [inline website/layout.py]: Just to clarify which file file it is that needs editing. 
```suggestion
    # Edit that file to modify the text content at the top of the page.
```

**Review by dandavies99 — CHANGES_REQUESTED** [review]
This is all looking great - the separation of concerns makes sense and keeps each file a nice length (the bootstrap components can cause them to get a bit unwieldy otherwise.)

I have no suggestions to improve the code itself. I added a placeholder suggestion to the README on the structure of the template - seems like a good place to explain briefly what each file does. Feel free to leave this in as a placeholder for the future or to update it now.

On a similar note, I wonder whether a short, module-level docstring at the top of each file would be useful at this stage.

**Comment by AdrianDAlessandro** [comment]
Works for me!

I also thought a module-level docstring would be helpful. Maybe that and the detail in the README can be tackled at the same time. ie when doing documentation

**Comment by dandavies99** [comment]
Yeh good idea, we'll do the docstrings and README details as one.

**Pull request merged**